Tools/MiniBrowser/gtk/BrowserTab.c

```diff
@@ -146,6 +146,21 @@ static void loadChanged(WebKitWebView *webView, WebKitLoadEvent loadEvent, Brows
     if (loadEvent != WEBKIT_LOAD_STARTED)
         return;
 
+    WebKitUserContentManager *userContentManager = webkit_web_view_get_user_content_manager(webView);
```

Comment:
This will run a few times for each top-level page load event except `WEBKIT_LOAD_STARTED` (skipped above). While this is not much of an issue (re-registering repeatedly should be idempotent) it is unneeded busy-work, I think it would be better to only do this during `WEBKIT_LOAD_COMMITTED` because at that point the URI of the page being loaded is already known to be definitive and will not change again for the current load.

Reply:
> This will run a few times for each top-level page load event except WEBKIT_LOAD_STARTED (skipped above)

I'm not sure I follow. Isn't the code above skipping every load-changed event except `WEBKIT_LOAD_STARTED`? So the code is only run for `WEBKIT_LOAD_STARTED`, which only once per top-level page, not the other way around :thinking:

Reply:
@cadubentzen Ah, you're right, sorry about the confusion. Nevertheless, it would be still better to use `_LOAD_COMMITTED` for checks on the URI, because it may change after `_LOAD_STARTED`, for example due to redirections.

Reply:
Alright, makes sense then! Will do, thanks :)